### ISSUE TITLE
[JENKINS-66490] ERROR trying to upgrade jenkins to 2.303.1 yields message: daemonize is needed

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#12152]
<p>hi,</p>

<p> </p>

<p>i'm trying to upgrade jenkins using the redhat package from:</p>

<p><a href="http://mirrors.jenkins-ci.org/redhat-stable/jenkins-2.303.1-1.1.noarch.rpm" class="external-link" target="_blank" rel="nofollow noopener">http://mirrors.jenkins-ci.org/redhat-stable/jenkins-2.303.1-1.1.noarch.rpm</a></p>

<p> </p>

<p>the following set of commands yields an installation error:</p>

<p>================================</p>

<p>-bash-4.2$ sudo rpm -Uvh jenkins-2.303.1-1.1.noarch.rpm<br/>
warning: jenkins-2.303.1-1.1.noarch.rpm: Header V4 RSA/SHA512 Signature, key ID 45f2c3d5: NOKEY<br/>
error: Failed dependencies:<br/>
 daemonize is needed by jenkins-2.303.1-1.1.noarch</p>

<p>================================</p>

<p> </p>

<p>for the past couple of years (at least 5 years) I was able to do so.</p>

<p>did something change in this rpm?</p>

<p>P.S. I installed openjdk11 a week ago - the jenkins controller is working properly.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/amidar.jpg" title="amidar's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar">amidar</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490" target="_blank">ERROR trying to upgrade jenkins to 2.303.1 yields message: daemonize is needed</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite">markewaite</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>label(s)</b>: OpenJDK, centos, core, install, java11, update, upgrade
<li><b>resolution</b>: Not A Defect
<li><b>resolved</b>: 2021-08-31T21:01:19+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>hi,</p>

<p> </p>

<p>i'm trying to upgrade jenkins using the redhat package from:</p>

<p><a href="http://mirrors.jenkins-ci.org/redhat-stable/jenkins-2.303.1-1.1.noarch.rpm" class="external-link" target="_blank" rel="nofollow noopener">http://mirrors.jenkins-ci.org/redhat-stable/jenkins-2.303.1-1.1.noarch.rpm</a></p>

<p> </p>

<p>the following set of commands yields an installation error:</p>

<p>================================</p>

<p>-bash-4.2$ sudo rpm -Uvh jenkins-2.303.1-1.1.noarch.rpm
warning: jenkins-2.303.1-1.1.noarch.rpm: Header V4 RSA/SHA512 Signature, key ID 45f2c3d5: NOKEY
error: Failed dependencies:
 daemonize is needed by jenkins-2.303.1-1.1.noarch</p>

<p>================================</p>

<p> </p>

<p>for the past couple of years (at least 5 years) I was able to do so.</p>

<p>did something change in this rpm?</p>

<p>P.S. I installed openjdk11 a week ago - the jenkins controller is working properly.</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
centos 7<br/>
openjdk11.0.12<br/>
working jenkins 2.289.2
```
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR-4532 - upgrade guide addition for new rpm dependency](https://github.com/jenkins-infra/jenkins.io/pull/4532)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412947&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412947">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Refer to the installation instructions at <a href="https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos</a> for the current install instructions.  The Jenkins rpm package now requires the <tt>daemonize</tt> package that is available from the epel-release repository.</p>

<p>Thanks for detecting that we didn't include that change in the Jenkins 2.303.1 upgrade guide.  It needs to be included there, in addition to the inclusion in the installation instructions.</p>
</pre>
</details>

</ul>
</details>

<p>Refer to the installation instructions at <a href="https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos" class="external-link" target="_blank" rel="nofollow noopener">https://www.jenkins.io/doc/book/installing/linux/#red-hat-centos</a> for the current install instructions.  The Jenkins rpm package now requires the <tt>daemonize</tt> package that is available from the epel-release repository.</p>

<p>Thanks for detecting that we didn't include that change in the Jenkins 2.303.1 upgrade guide.  It needs to be included there, in addition to the inclusion in the installation instructions.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/amidar.jpg" title="amidar's avatar" /> amidar:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412951&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412951">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>!</p>

<p>since I'm installing it on an air-gapped centos 6 server, can you please make sure the instructions contain exactly what is needed - package wise?</p>

<p>also - is there any additional configuration which needs to be done prior to the upgrade itself.</p>

<p>thanks.</p>
</pre>
</details>

</ul>
</details>

<p>thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a>!</p>

<p>since I'm installing it on an air-gapped centos 6 server, can you please make sure the instructions contain exactly what is needed - package wise?</p>

<p>also - is there any additional configuration which needs to be done prior to the upgrade itself.</p>

<p>thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412952&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412952">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar" class="user-hover" rel="amidar">amidar</a> I'm sorry, but I don't test CentOS 6 and don't intend to document or test for CentOS 6.  CentOS 6 <a href="https://wiki.centos.org/About/Product" class="external-link" target="_blank" rel="nofollow noopener">end of life</a> was November 2020.</p>

<p>The command line git version included in CentOS 6 has several use cases that do not work with the Jenkins git plugin.  CentOS 6 does not receive maintenance updates.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar" class="user-hover" rel="amidar">amidar</a> I'm sorry, but I don't test CentOS 6 and don't intend to document or test for CentOS 6.  CentOS 6 <a href="https://wiki.centos.org/About/Product" class="external-link" target="_blank" rel="nofollow noopener">end of life</a> was November 2020.</p>

<p>The command line git version included in CentOS 6 has several use cases that do not work with the Jenkins git plugin.  CentOS 6 does not receive maintenance updates.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/amidar.jpg" title="amidar's avatar" /> amidar:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412953&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412953">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> fair enough.</p>

<p> </p>

<p>how about CentOS 7? </p>

<p>I guess the time for server upgrade has come... still I need to know how to treat this air gapped environment... (up until now the jenkins rpm was all I needed...)</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a> fair enough.</p>

<p> </p>

<p>how about CentOS 7? </p>

<p>I guess the time for server upgrade has come... still I need to know how to treat this air gapped environment... (up until now the jenkins rpm was all I needed...)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412954&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412954">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar" class="user-hover" rel="amidar">amidar</a> The existing instructions work for CentOS 7 with an internet connection.  You'll need to explore for yourself to see what is required in your air-gapped environment.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=amidar" class="user-hover" rel="amidar">amidar</a> The existing instructions work for CentOS 7 with an internet connection.  You'll need to explore for yourself to see what is required in your air-gapped environment.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/phreakadelle.png" title="phreakadelle's avatar" /> phreakadelle:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412989&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412989">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Same problem here on one of our old Jenkins Boxes with CentOS 6 (which we still have in place for some legacy applications).</p>

<p>So far i only installed daemonize manually</p>
<ul>
	<li><a href="http://software.clapper.org/daemonize/" class="external-link" target="_blank" rel="nofollow noopener">http://software.clapper.org/daemonize/</a></li>
</ul>


<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">wget https:<span class="code-comment">//github.com/bmc/daemonize/archive/refs/heads/master.zip
</span>unzip master.zip
cd daemonize-master/
configure
make
make install
</pre>
</div></div>

<p>But havent solved it yet... </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">[root@swpsci01 tmp]# rpm -U jenkins-latest.rpm
warning: jenkins-latest.rpm: Header V4 RSA/SHA512 Signature, key ID 45f2c3d5: NOKEY
error: Failed dependencies:
        daemonize is needed by jenkins-2.303.1-1.1.noarch
[root@swpsci01 tmp]# daemonize
daemonize, version 1.7.8

</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>Same problem here on one of our old Jenkins Boxes with CentOS 6 (which we still have in place for some legacy applications).</p>

<p>So far i only installed daemonize manually</p>
<ul>
	<li><a href="http://software.clapper.org/daemonize/" class="external-link" target="_blank" rel="nofollow noopener">http://software.clapper.org/daemonize/</a></li>
</ul>



<pre>
wget https:<span class="code-comment">//github.com/bmc/daemonize/archive/refs/heads/master.zip
</span>unzip master.zip
cd daemonize-master/
configure
make
make install
</pre>

<p>But havent solved it yet... </p>


<pre>
[root@​swpsci01 tmp]# rpm -U jenkins-latest.rpm
warning: jenkins-latest.rpm: Header V4 RSA/SHA512 Signature, key ID 45f2c3d5: NOKEY
error: Failed dependencies:
daemonize is needed by jenkins-2.303.1-1.1.noarch
[root@​swpsci01 tmp]# daemonize
daemonize, version 1.7.8

</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=412995&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-412995">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=phreakadelle" class="user-hover" rel="phreakadelle">phreakadelle</a> the dependency in the rpm file can only be satisfied with an rpm package for daemonize.  Installing from source won't do it because the RPM package manager won't realize that the dependency is satisfied.  You might try enabling the epel-release repository (if they still support CentOS 6) and then installing.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=phreakadelle" class="user-hover" rel="phreakadelle">phreakadelle</a> the dependency in the rpm file can only be satisfied with an rpm package for daemonize.  Installing from source won't do it because the RPM package manager won't realize that the dependency is satisfied.  You might try enabling the epel-release repository (if they still support CentOS 6) and then installing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/amidar.jpg" title="amidar's avatar" /> amidar:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=413002&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-413002">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I eventually updated the air gapped Jenkins server on our site. it has centos 6.4 and I used the following packages:</p>

<p><a href="https://fedora-archive.ip-connect.info/epel/6/x86_64/epel-release-6-8.noarch.rpm" class="external-link" target="_blank" rel="nofollow noopener">https://fedora-archive.ip-connect.info/epel/6/x86_64/epel-release-6-8.noarch.rpm</a></p>

<p><a href="https://fedora-archive.ip-connect.info/epel/6/x86_64/Packages/d/daemonize-1.7.3-1.el6.x86_64.rpm" class="external-link" target="_blank" rel="nofollow noopener">https://fedora-archive.ip-connect.info/epel/6/x86_64/Packages/d/daemonize-1.7.3-1.el6.x86_64.rpm</a></p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>I eventually updated the air gapped Jenkins server on our site. it has centos 6.4 and I used the following packages:</p>

<p><a href="https://fedora-archive.ip-connect.info/epel/6/x86_64/epel-release-6-8.noarch.rpm" class="external-link" target="_blank" rel="nofollow noopener">https://fedora-archive.ip-connect.info/epel/6/x86_64/epel-release-6-8.noarch.rpm</a></p>

<p><a href="https://fedora-archive.ip-connect.info/epel/6/x86_64/Packages/d/daemonize-1.7.3-1.el6.x86_64.rpm" class="external-link" target="_blank" rel="nofollow noopener">https://fedora-archive.ip-connect.info/epel/6/x86_64/Packages/d/daemonize-1.7.3-1.el6.x86_64.rpm</a></p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-66490?focusedId=413007&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-413007">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Fixed by updating the documentation to mention the change in RPM package definition</p>
</pre>
</details>

</ul>
</details>

<p>Fixed by updating the documentation to mention the change in RPM package definition</p>
